lib/iris/coords.py

```diff
@@ -2921,6 +2921,24 @@ def xml_element(self, doc):
         return element
 
 
+_dim_coord_cache = {}
```

Comment:
Should I enforce a maximum cache size?

Reply:
(applies more widely)

Reply:
In places we use no limit, in others the `lru_cache` default of 128. I've added 128 here with a comment